superdesk/publish/publish_queue.py

```diff
@@ -116,6 +137,43 @@ def _set_queue_state(self, updates, original):
             else updates.get("state") or QueueState.PENDING.value
         )
 
+    def is_updated_document(self, doc):
+        if not doc.get("item_id"):
+            return
+        article = get_resource_service("published").find_one(req=None, guid=doc["item_id"])
+        if not article:
+            return
+        return article.get("rewrite_of")
+
+    def is_corrected_document(self, doc):
+        if not doc.get("item_id"):
+            return
+        article = get_resource_service("published").find_one(req=None, guid=doc["item_id"])
+        if not article:
+            return
+        return doc.get("publishing_action") == "corrected" or article.get("operation") == "being_corrected"
+
+    def resend_association_items(self, doc):
+        try:
+            associated_items = (
+                json.loads(doc.get("formatted_item", {})).get("associations", {}) if doc.get("formatted_item") else {}
+            )
+        except Exception:
+            return
+
+        if associated_items:
+            for key, associated_value in associated_items.items():
+                associated_article = get_resource_service("published").find_one(
+                    req=None, guid=associated_value.get("guid")
+                )
+                subscriber = get_resource_service("subscribers").find_one(req=None, _id=doc["subscriber_id"])
```

Comment:
the subscriber where the doc is sent might not actually be the same where the media goes, in case of cp they have one subscriber for text items and another for media, the idea is to republish it when the text item is published and send it whereever it should go based on products/subscribers